plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,197 @@
+
+    final static JSONObject schemaTemplateObject = new JSONObject()
+        .put("$schema", "http://json-schema.org/draft-07/schema#")
+        .put("description", "Jenkins Configuration as Code")
+        .put("type", "object");
+
+    public static JSONObject generateSchema() {
+
+        /**
+         * The initial template for the JSON Schema
+         */
+        JSONObject schemaObject = new JSONObject(schemaTemplateObject.toString());
+        /**
+         * This generates the schema for the root configurators
+         * Iterates over the root elements and adds them to the schema.
+         */
+        schemaObject.put("properties", generateRootConfiguratorObject());
+        /**
+         * This generates the schema for the base configurators
+         * Iterates over the base configurators and adds them to the schema.
+         */
+        JSONObject schemaConfiguratorObjects = new JSONObject();
+        ConfigurationAsCode configurationAsCode = ConfigurationAsCode.get();
+        for (Object configuratorObject : configurationAsCode.getConfigurators()) {
+            if (configuratorObject instanceof BaseConfigurator) {
+                BaseConfigurator baseConfigurator = (BaseConfigurator) configuratorObject;
+                List<Attribute> baseConfigAttributeList = baseConfigurator.getAttributes();
+                if (baseConfigAttributeList.size() == 0) {
+                    schemaConfiguratorObjects
+                        .put(((BaseConfigurator) configuratorObject).getTarget().getSimpleName().toLowerCase(),
+                            new JSONObject()
+                                .put("type", "object")
+                                .put("properties", new JSONObject()));
+
+                } else {
+                    JSONObject attributeSchema = new JSONObject();
+                    for (Attribute attribute : baseConfigAttributeList) {
+                        if (attribute.multiple) {
+                            generateMultipleAttributeSchema(attributeSchema, attribute);
+                        } else {
+                            if (attribute.type.isEnum()) {
+                                generateEnumAttributeSchema(attributeSchema, attribute);
+                            } else {
+                                attributeSchema.put(attribute.getName(), generateNonEnumAttributeObject(attribute));
+                                schemaConfiguratorObjects
+                                    .put(((BaseConfigurator) configuratorObject).getTarget().getSimpleName().toLowerCase(),
+                                        new JSONObject()
+                                            .put("type", "object")
+                                            .put("properties", attributeSchema));
+                            }
+                        }
+                    }
+                }
+            }
+            /**
+             * Used to generate the schema for the implementors of
```

Comment:
Is this comment necessary anymore? The instance of check is pretty clear

Reply:
Yeah got rit of it.